### PR TITLE
Ad login

### PR DESCRIPTION
**Pull request by setseed42**

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/src/security.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    if access_token:
+        user_info = get_user_info(access_token)
+        st.session_state['user_info'] = user_info
+        return True
+    else:
+        st.write("Please sign-in to use this app.")
+        auth_url = get_auth_url(app)
+        st.markdown(
+            f"<a href='{auth_url}' target='_self'>Sign In</a>", unsafe_allow_html=True)
+        st.stop()
```
Comment (e-larsson) [inline {{ cookiecutter.repo_name }}/src/security.py:67]: Is this rendered as text? Would it be easy to switch to a Login button?

File: {{ cookiecutter.repo_name }}/src/security.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            st.session_state['access_token'] = access_token
+            st.experimental_set_query_params()
+
+
+def login_step():
+    CLIENT_ID = os.environ['CLIENT_ID']
+    CLIENT_SECRET = os.environ['CLIENT_SECRET']
+    TENANT_ID = os.environ['TENANT_ID']
+
+    AUTHORITY = f'https://login.microsoftonline.com/{TENANT_ID}'
```
Comment (e-larsson) [inline {{ cookiecutter.repo_name }}/src/security.py:45]: If we want to run GSuite instead, what would need to be changed? (Let's not prepare too much for that case right now, I'm just interested)

File: {{ cookiecutter.repo_name }}/.github/workflows/ci-cd.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
```
Comment (e-larsson) [inline {{ cookiecutter.repo_name }}/.github/workflows/ci-cd.yaml:20]: Why not Python 3.11?

**Review by e-larsson — CHANGES_REQUESTED** [review]
Looks good! 🙌 Very clean, and neat. Just had some questions

File: {{ cookiecutter.repo_name }}/.github/workflows/ci-cd.yaml (hunk 3 above)
Comment (setseed42) [reply {{ cookiecutter.repo_name }}/.github/workflows/ci-cd.yaml]: Fixed

**Pull request merged**